src/main/java/org/beanpod/switchboard/config/CustomAuthenticationEntryPoint.java

```diff
@@ -0,0 +1,37 @@
+package org.beanpod.switchboard.config;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.io.IOException;
+import java.io.PrintWriter;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.web.AuthenticationEntryPoint;
+import org.springframework.stereotype.Component;
+
+@Component
+public class CustomAuthenticationEntryPoint implements AuthenticationEntryPoint {
```

Comment:
What this does is leverage the spring security framework to send a JSON response when a user is unauthorized. For example, if a client with the role USER tries to do admin functions such as create another user, then instead of getting the generic 403 FORBIDDEN response, I customized it to return a 401 UNAUTHORIZED response.

403 FORBIDDEN is still used for non-users who try to access the server.

Reply:
I don't know if you'll agree, but `CustomAuthenticationEntryPoint` is more appropriate than your suggestion?

Reply:
I see, let's go with what we have now.